src/main/python/api.py

```diff
@@ -93,8 +99,7 @@ def post(self, payload: dict = {}):
             if res_json.get('token'):
                 self.auth = True
                 self.token = res_json.get('token')
-                if 'login' in res_json.get('message', '').lower():
-                    self.store.put(self.token)
+            self._set_state()
```

Comment:
If `post` is the only place need to update `self.store`, we can remove the call from here.

Reply:
I'm not sure I follow... `_set_state()` updates the store's state, so it's definitely needed here.

Reply:
I am not pretty sure if we need to update `self.store` after each api call. Based on the logic I saw, when we open `Api` with `with` syntax, it will call `__enter__`, open `.credential_store` and read whatever in there into `self.token`. Then if we call `post` function, it will do `self.token = res_json.get('token')` to update the value it holds. However, if we call `get` function, it will just do nothing, but just write the value back to the file.

Reply:
Well, not exactly. We would have to call the the `post` and `get` functions on the same `Api(...)` instance. Writing to the store every time we get a token, or logout of the application ensures we don't run into problems where we have a stale token leftover after logouts and we have a token stored to automatically login again.

Reply:
I don't think we will have a the same `Api` instance. As soon as the code inside the `with` block is executed, all the acquired resources are released. Basically, as I understand, everytime we have a `api` call (get request, post request, etc), we do something, and when it is done, the `api` object would be release. Therefore, if we do a `get` action, and we will just read token from `.credential_store` and write it back (without any change).

Reply:
That's mostly correct but we only write the token back when we've received a new one from the API. I.e. when we hit `/auth/refresh` or `/auth/login`.